### PR TITLE
Boilerplate code from the guide

### PR DESCRIPTION
**Pull request by espen42**
Main focus is the .js and .jsx files.

**Context:** The "boilerplate" branch is intended to mirror the "copy-this-and-youre-ready-to-go" code in the Next.js guide (also WiP). It should be possible for the guide to refer to cloning that branch, for those who don't want to actually copy-paste (it's a bit much code, especially fetchContent.js). 

So the code here is **simplified** for readability, minimized copy-pasting, and a clean-slate boilerplate state. No typescript, so some of the code is simply removing that / renaming .ts and .tsx to .js and .jsx

("boilerplate" will be different from the master branch, which should be less minimized and more state-of-the-art: typescript, better division into modules, `buildFetcher` with typeSelector and config parameters (which allows extracting `fetchContent` to a standalone NPM package)...)

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,3 +1,8 @@
 {
-  "extends": ["next", "next/core-web-vitals"]
+  "extends": ["next", "next/core-web-vitals"],
+    "rules": {
+        // Other rules
+        "@next/next/no-img-element": "off"
```
Comment (poi33) [inline .eslintrc:5]: Hmm custom eslint rules for people who use our starter :D

File: src/components/BasePage.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    if (!content) {
+        console.warn("No 'content' data in BasePage props");
+        return null;
+    }
+
+    if (!meta || !meta.type) {
+        console.warn("BasePage props are missing 'meta.type'. Falling back to DefaultPage component.");
+    }
+
+    const typeSelection = (typeSelector || {})[meta.type]
```
Comment (poi33) [inline src/components/BasePage.jsx:62]: use default param instead? `meta={}`

File: src/pages/_app.jsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/**
+ *
+ * @param pageProps {{content, meta, error}}
```
Comment (poi33) [inline src/pages/_app.jsx:3]: The jsdoc is about "MyApp" 
Needs to include Component too. And maybe a short "Main roo element" or simular

**Review by poi33 — CHANGES_REQUESTED** [review]

File: .eslintrc (hunk 1 above)
Comment (espen42) [reply .eslintrc:5]: Needed for  tags to be allowed from next.js, instead of enforcing next's <Image> 
:(

**Review by poi33 — APPROVED** [review]

**Pull request merged**